common/txmgr/types/head_trackable.go

```diff
@@ -0,0 +1,12 @@
+package types
+
+import "context"
+
+// HeadTrackable represents a generic type, implemented by the core txm,
+// to be able to receive head events from any chain.
+// Chain implementations should notify head events to the core txm via this interface.
+//
+//go:generate mockery --quiet --name HeadTrackable --output ../mocks/ --case=underscore
+type HeadTrackable[HEAD any] interface {
+	OnNewLongestChain(ctx context.Context, head HeadView[HEAD])
```

Comment:
WDTY about simplifying this irregular name while we're changing things here?
```suggestion
	NewHead(ctx context.Context, head HeadView[HEAD])
```

Reply:
This is indicating that we found a head which creates a new longest chain, not just any head.
For example, if we had 2 forks, A and B, like this:
A: 4,5,6,7
B:4,5

Now if we see a new head #6 on B fork, it won't trigger this callback. Only trigger on A8, or if we reach B8 before A8.

Reply:
See this for more context: https://github.com/smartcontractkit/chainlink/blob/e7527e6048e9712e71921973b85c731b4cf54113/core/chains/evm/headtracker/head_tracker.go#L203-L211

Reply:
How about `NewHighestHead` like in that error message?
```suggestion
	NewHighestHead(ctx context.Context, head HeadView[HEAD])
```

Reply:
Yes, I would prefer that name. However note that OnLongestChain name is used in all places that subscribe to HeadBroadcaster, which is way more places than just TxMgr.
If we want to do that, I would prefer to do a repo wide rename in a dedicated PR.